builtins: add test that checks all spatial functions handle POINT EMPTY

otan opened the pull request:
SQLSmith can take a while to surface some things, so let's add this test
that catches a lot of people offguard immediately.

Release note: None

sumeerbhola committed:
buitins,geomfn: bug fix in ST_Scale when factor, origin are empty points

Fixes #54603

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/54887)
<!-- Reviewable:end -->

otan committed:
builtins: add test that checks all spatial functions handle POINT EMPTY

SQLSmith can take a while to surface some things, so let's add this test
that catches a lot of people offguard immediately.

Release note: None

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54887#-:-MIQYLQpA99Y13vHu8vS:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=sumeerbhola
